src/components/CategoryFilter.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
```

Comment:
While the instructions "probably" implied/mentioned that we should pass a `handleFilterChange` prop here from `BookList`, a better way, actually is just connect CategoryFilter to the `redux` store:

```javascript
import { connect } from 'react-redux'
import { changeFilter } from '../actions/index';

const CategoryFilter = ({ changeFilter }) => {
  // onChange part:
 <select
          id="filter"
          onChange={e => {
            changeFilter(e.target.value);
          }}
        >
} 

export default connect(null, { changeFilter })(CategoryFilter);
```
As much as we can avoid passing of props from the parent component, the better. This makes our components "reusable" (a good way to architecture React apps).